tests/db2/test-db2.cpp

```diff
@@ -20,6 +20,17 @@ using namespace soci::tests;
 std::string connectString;
 backend_factory const &backEnd = *soci::factory_db2();
 
+// It appears later versions of GCC arent happy with this - to be fixed properly
+
+#if (__GNUC__ == 4 && (__GNUC_MINOR__ > 6)) || (__clang__ == 1)
+namespace boost {
+    std::basic_ostream<char, std::char_traits<char> >& operator<< (std::basic_ostream<char, std::char_traits<char> > & stream, boost::optional<int> const & value){
```

Comment:
I'm still unclear myself, so will have to look again before I know what's the problem :)